File: src/crawler/tiki/parser.py
# parser.py
def parse_tiki_product(item, keyword):
    product_id = item.get("id")
    if not product_id:
        return None

    price = item.get("price", 0)
    original_price = item.get("original_price", 0)
    discount_percent = (
        round((original_price - price) / original_price * 100, 2)
        if original_price else 0
    )

    return {
        "platform": "Tiki",
        "product_id": str(product_id),
        "product_name": item.get("name", "").strip(),
        "price": price,
        "original_price": original_price,
        "discount_percent": discount_percent,
        "product_url": "https://tiki.vn/" + (item.get("url_path") or ""),
        "image_url": item.get("thumbnail_url"),
        "rating": item.get("rating_average", 0),
        "review_count": item.get("review_count", 0),
        "category": keyword
    }

File: src/crawler/Crawl_cellphones/parser.js
/**
 * Transform dữ liệu từ CellphoneS GraphQL API sang format JSONL
 */
export function transformCellphonesData(product, category) {
    const price = parseFloat(product.special_price || product.price || 0);
    const originalPrice = parseFloat(product.price || 0);

    // Tính discount_percent nếu có giá gốc
    let discountPercent = 0;
    if (originalPrice > price && originalPrice > 0) {
        discountPercent = Math.round(((originalPrice - price) / originalPrice) * 100);
    }

    return {
        platform: 'cellphones',
        product_id: String(product.product_id || ''),
        product_name: product.name || '',
        price: price,
        original_price: originalPrice,
        discount_percent: discountPercent,
        product_url: product.url_path
            ? `https://cellphones.com.vn/${product.url_path}.html`
            : '',
        image_url: product.thumbnail || '',
        rating: 0, // API không cung cấp field này
        review_count: 0, // API không cung cấp field này
        category: category || ''
    };
}
